aws-servicebroker/Makefile

```diff
@@ -1,54 +1,104 @@
+		--key-schema \
+			AttributeName=id,KeyType=HASH \
+			AttributeName=userid,KeyType=RANGE \
+		--global-secondary-indexes \
+			'IndexName=type-userid-index,KeySchema=[{AttributeName=type,KeyType=HASH},{AttributeName=userid,KeyType=RANGE}],Projection={ProjectionType=INCLUDE,NonKeyAttributes=[id,userid,type,locked]},ProvisionedThroughput={ReadCapacityUnits=5,WriteCapacityUnits=5}' \
+		--provisioned-throughput \
+			ReadCapacityUnits=5,WriteCapacityUnits=5 \
+		--region ${REGION} --table-name awssb
+	aws dynamodb wait table-exists --table-name awssb --region ${REGION}
 
-build-init:
-	docker build -t ${DOCKER_PREFIX}${DOCKER_REPOSITORY}/aws-service-broker-init:${DOCKER_TAG} -f Dockerfile-init .
+undeploy:
+	-helm delete --purge aws
+	-aws dynamodb delete-table --table-name awssb --region ${REGION}
 
-publish-init:
-	docker push ${DOCKER_PREFIX}${DOCKER_REPOSITORY}/aws-service-broker-init:${DOCKER_TAG}
+ca.key:
+	kubectl get secret -n ${CF_NAMESPACE} -o jsonpath='{.items[*].data.internal-ca-cert-key}' | base64 -di > $@
 
-deploy:
-	helm ${COMMAND} --namespace ${NAMESPACE} \
-		--set image.registry.hostname="${REGISTRY_HOSTNAME}" \
-		--set image.registry.username="${REGISTRY_USERNAME}" \
-		--set image.registry.password="${REGISTRY_PASSWORD}" \
-		--set image.organization=${DOCKER_REPOSITORY} \
-		--set image.tag=${DOCKER_TAG} \
-		--set image.pullPolicy=Always \
-		--set aws.insecure=true \
-		--set aws.keyid="$$(aws configure get aws_access_key_id)" \
+ca.pem:
+	kubectl get secret -n ${CF_NAMESPACE} -o jsonpath='{.items[*].data.internal-ca-cert}' | base64 -di > $@
+
+tls.key.encrypted:
+	openssl req -newkey rsa:4096 -keyout tls.key.encrypted -out tls.req -days 365 \
+		-passout pass:1234 \
+		-subj '/CN=aws-servicebroker.${NAMESPACE}' -verbose -batch \
+		</dev/null
+
+tls.req: tls.key.encrypted
+	# built as a side effect
+
+tls.key: tls.key.encrypted
+	openssl rsa -in $< -passin pass:1234 -out $@
+
+tls.pem: tls.req ca.key ca.pem
+	openssl x509 -req -CA ca.pem -CAkey ca.key -CAcreateserial \
+		-in tls.req -out tls.pem
+	-rm -f ca.srl
+
+.INTERMEDIATE: tls.pem tls.key tls.req tls.key.encrypted ca.pem ca.key ca.srl
+
+deploy: tls.pem tls.key
+	helm install \
+		--repo https://awsservicebroker.s3.amazonaws.com/charts \
+		aws-servicebroker \
+		--version ${BROKER_VERSION} \
+		--name aws \
+		--namespace aws \
+		--set deployClusterServiceBroker=false \
+		--set authenticate=false \
+		--set aws.accesskeyid="$$(aws configure get aws_access_key_id)" \
 		--set aws.secretkey="$$(aws configure get aws_secret_access_key)" \
-		--set aws.rolearn="${IAM_ROLE}" \
-		--set broker.username="${BROKER_USERNAME}" \
-		--set broker.password="${BROKER_PASSWORD}" \
-		--debug \
-		$(if $(filter install,${COMMAND}), --dry-run --name) aws ./helm
+		--set aws.region=${REGION} \
+		--set tls.cert="$$(base64 --wrap=0 tls.pem)" \
+		--set tls.key="$$(base64 --wrap=0 tls.key)" \
+		--set-string aws.targetaccountid=138384977974 \
```

Comment:
Is this `id` something SUSE specific ? I am wondering if this can change, and/or also if this should be public at all ?!

Reply:
You're right, let me go fix that. (It's us-specific)

Reply:
Looks good.